service/TransactionsServiceImpl.java

```diff
@@ -37,48 +61,70 @@ public StatisticsDTO getStatistics() {
-        LOGGER.debug("CURRENT DATE TIME {}",currentDateTime);
+        LOGGER.debug("LIMIT CURRENT DATE TIME {} size list {} ", limitDateTime, transactions.size());
 
         /* GET VALID TRANSACTIONS valid transaction= from 60 seg. to current time */
-        if(Objects.nonNull(transactions) && !transactions.isEmpty()) {
-            int i = 0;
-            while (i < transactions.size() && transactions.get(i).getDateTime().isAfter(limitDateTime) &&
-                    transactions.get(i).getDateTime().isBefore(currentDateTime)) {
-
-                sum = sum.add(transactions.get(i).getAmount());
-                max = max.max(transactions.get(i).getAmount());
-                if (i == 0) {
-                    min = transactions.get(i).getAmount();
-                } else {
-                    min = min.min(transactions.get(i).getAmount());
-                }
-                i++;
-            }
 
-            count = Long.valueOf(i);
-            BigDecimal divide = new BigDecimal(count);
-            avg = sum.divide(divide,2,RoundingMode.HALF_UP);
 
-            /* DELETE NO VALID TRANSACTIONS */
-            transactions=transactions.subList(0,i-1);
-        }
+        synchronized (this) {
+            AtomicInteger i = new AtomicInteger(0);
+            if (Objects.nonNull(transactions) && !transactions.isEmpty()) {
+                Iterator<Transaction> iterator = transactions.iterator();
 
-        return new StatisticsDTO(sum.setScale(2,RoundingMode.HALF_UP).toString(), avg.setScale(2,RoundingMode.HALF_UP).toString(),
-                max.setScale(2,RoundingMode.HALF_UP).toString(), min.setScale(2,RoundingMode.HALF_UP).toString(), count);
+                while (iterator.hasNext()) {
+                    Transaction trans = iterator.next();
 
-    }
+                    sum = sum.add(trans.getAmount());
+                    max = max.max(trans.getAmount());
+                    if (i.get() == 0) {
+                        min = trans.getAmount();
+                    } else {
+                        min = min.min(trans.getAmount());
+                    }
+                    i.incrementAndGet();
 
-    @Override
-    public void saveTransaction(TransactionDTO transactionDTO) throws NotFoundException, UnprocessableException {
-        ValidationUtils.validateTransaction(transactionDTO);
-        transactions.add(new Transaction(transactionDTO));
-        //sort the list
-        transactions=transactions.stream().sorted(Comparator.comparing(Transaction::getDateTime).reversed()).collect(Collectors.toList());
+                }
+                if (i.get() > 0) {
+                    count = Long.valueOf(i.get());
+                    BigDecimal divide = new BigDecimal(count);
+                    avg = sum.divide(divide, 2, RoundingMode.HALF_UP);
+                }
+
+
+            }
+
+            statisticsDTO.setSum(sum.setScale(2, RoundingMode.HALF_UP).toString());
```

Comment:
`RoundingMode.HALF_UP).toString()` repeated four lines, this is verbose. Can you make a utility?